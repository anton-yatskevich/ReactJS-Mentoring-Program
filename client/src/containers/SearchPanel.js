import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SearchFieldSelect from '../components/SearchFieldSelect';

class SearchPanel extends Component {
    constructor(props) {
        super(props);
        this.state = { value: '' };
        this.onSubmit = props.onSubmit;
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({ value: event.target.value });
    }

    handleSubmit(event) {
        event.preventDefault();
        const { value } = this.state;
        this.onSubmit(value.toLowerCase());
    }

    render() {
        const { value } = this.state;
        const { onChangeSearchField, searchField } = this.props;

        return (
            <form onSubmit={this.handleSubmit}>
                <label htmlFor="searchInput">
                    Find your movie
                    <input type="text" id="searchInput" value={value} onChange={this.handleChange} />
                </label>
                <SearchFieldSelect
                    onChangeSearchField={onChangeSearchField}
                    searchField={searchField}
                />
                <input type="submit" value="Submit" />
            </form>
        );
    }
}

SearchPanel.propTypes = {
    searchField: PropTypes.string,
    onSubmit: PropTypes.func,
    onChangeSearchField: PropTypes.func
};

SearchPanel.defaultProps = {
    searchField: null,
    onSubmit: null,
    onChangeSearchField: null
};

export default SearchPanel;
