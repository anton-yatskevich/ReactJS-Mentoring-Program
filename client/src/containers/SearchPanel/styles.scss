.search-panel {
	width: 100%;
	background-color: rgb(146, 133, 133);
	padding: 40px 20px;
	color: #fff;

	&__form {
		max-width: 1200px;
		margin: 0 auto;
	}

	&__title {
		font-size: 30px;
		text-transform: uppercase;
	}

	&__input {
		width: 100%;    
		font-size: 20px;
		padding: 6px 10px;
		margin: 10px 0;
	}

	&__controls {
		display: flex;
		justify-content: space-between;
		align-items: center;
	}

	&__button--submit {
		padding: 8px 30px;
		font-size: 18px;
		text-transform: uppercase;
		background-color: rgb(160, 0, 0);
		border: none;
		color: white;
		text-decoration: none;
		cursor: pointer;
	}

	&__button {
		padding: 5px 20px;
		font-size: 14px;
		text-transform: uppercase;
		background-color: rgb(173, 173, 173);
		border: none;
		color: white;
		text-decoration: none;
		margin-left: 15px;
		transition: all 0.5;
		cursor: pointer;

		&--active {
			background-color: rgb(160, 0, 0);
		}
	}

	&__field-select-title {
		text-transform: uppercase;
		font-size: 14px;
	}
}