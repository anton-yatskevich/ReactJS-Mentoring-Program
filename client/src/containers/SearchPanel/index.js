import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import noop from 'lodash.noop';
import fetchMovies from '../../actions/fetchMovies';
import SearchFieldSelect from '../SearchField';
import './styles.scss';

export class SearchPanelComponent extends Component {
    state = { value: '' };

    handleChange = (event) => {
        this.setState({ value: event.target.value });
    }

    handleSubmit = (event) => {
        event.preventDefault();

        const { value } = this.state;
        const { getMovies } = this.props;

        getMovies(value.toLowerCase());
    }

    render() {
        const { value } = this.state;

        return (
            <div className="search-panel">
                <form className="search-panel__form" onSubmit={this.handleSubmit}>
                    <p className="search-panel__title">Find your movie</p>
                    <input className="search-panel__input" type="text" value={value} onChange={this.handleChange} placeholder="Search..." />
                    <div className="search-panel__controls">
                        <SearchFieldSelect />
                        <input className="search-panel__button--submit" type="submit" value="Search" />
                    </div>
                </form>
            </div>
        );
    }
}

SearchPanelComponent.propTypes = {
    getMovies: PropTypes.func
};

SearchPanelComponent.defaultProps = {
    getMovies: noop
};

export default connect(null, { getMovies: fetchMovies })(SearchPanelComponent);
