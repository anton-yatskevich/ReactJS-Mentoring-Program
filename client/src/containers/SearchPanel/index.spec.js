import React from 'react';
import { shallow } from 'enzyme';
import SearchPanel from './index';

describe('SearchPanel component', () => {
    let SearchPanelComponent;

    beforeAll(() => {
        SearchPanelComponent = shallow(<SearchPanel />);
    });

    it('should be rendered with default props', () => {
        expect(SearchPanelComponent).toMatchSnapshot();
    });

    it('should set new value if input value is cahnged', () => {
        const inputElement = SearchPanelComponent.find('.search-panel__input');
        inputElement.simulate('change', { target: { value: 'newValue' } });
        expect(SearchPanelComponent.state().value).toBe('newValue');
    });

    it('should set onSubmitCallback if form was submitted', () => {
        const preventMock = jest.fn();
        const submitMock = jest.fn();
        const props = {
            onSubmit: submitMock
        };
        const SearchPanelComponentWitHandler = shallow(<SearchPanel {...props} />);
        const Form = SearchPanelComponentWitHandler.find('.search-panel__form');
        Form.simulate('submit', {
            preventDefault: preventMock
        });
        expect(preventMock).toHaveBeenCalled();
        expect(submitMock).toHaveBeenCalled();
    });
});
