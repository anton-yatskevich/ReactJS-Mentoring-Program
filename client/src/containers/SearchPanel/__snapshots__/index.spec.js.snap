// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel component should be rendered with default props 1`] = `
<div
  className="search-panel"
>
  <form
    className="search-panel__form"
    onSubmit={[Function]}
  >
    <p
      className="search-panel__title"
    >
      Find your movie
    </p>
    <input
      className="search-panel__input"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <div
      className="search-panel__controls"
    >
      <SearchFieldSelect
        onChangeSearchField={[Function]}
        searchField=""
      />
      <input
        className="search-panel__button--submit"
        type="submit"
        value="Search"
      />
    </div>
  </form>
</div>
`;
